### PR TITLE
[Fixes #308] Publish code to Github

### LINKED ISSUE #308: Add an option to publish code to Github
## Feature Request

Is your feature request related to an existing [Issue](https://www.github.com/Auto-DL/Generator/issues) or a new implementation?

No

### Feature Description

After generating the code, add a button to allow the user to directly publish the generated code to the user's Github repository.

### PR DESCRIPTION
**Pull request by therajtiwari**
# Pull Request

## What does this PR do?
Fixes #308 

- Using can authorize with their GitHub account using OAuth
- Once authorized, after generating the code users can directly push the code to their GitHub account 
- Users can logout and switch to another account

Screenshot from 2021-08-30 18-39-39

Screenshot from 2021-09-03 20-03-09

Screenshot from 2021-09-03 20-02-25

Screenshot from 2021-09-03 20-02-36

Screenshot from 2021-09-03 20-11-38

## What part does this affect?
- [x] FrontEnd.
- [x] BackEnd.
- [ ] Documentation.
- [ ] Other. (Please specify below)

## Before submitting
- [x] Was this discussed/approved via a GitHub issue or slack?
- [x] Did you read the [contributor guideline](https://github.com/Auto-DL/Auto-DL/blob/v1-beta/CONTRIBUTING.md)?
- [x] Did you ensure that there aren't any other open [Pull Requests](https://www.github.com/Auto-DL/Generator/pulls) for the same update/change?
- [x] Did you make sure the title is self-explanatory and the description concisely explains the PR?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure the code is clean and docstrings have been added or updated as required?
- [x] Did you make sure the code is linted/formatted locally prior to submission? (using `black` and/or `prettier`)
- [ ] Did you make sure to update the documentation with your changes? (if necessary)

## PR review
Anyone in the community is free to review the PR once the tests have passed.

Thank you for contributing to [AutoDL](https://github.com/Auto-DL/Auto-DL). We look forward to your continued support.

### CODE REVIEW

**Comment by therajtiwari** [comment]
@ADI10HERO @rajraj889  conflicts have been fixed, logout feature has been added, and UI has been updated, please review it and let me know

**Review by rajraj889 — APPROVED** [review]
NIce work :)

File: FrontEndApp/v1-react/src/Components/MainApp/HomeService.js [hunk 1]
```diff
@@ -1,81 +1,105 @@
...
-      });
+      const response = await axios.post(
+        `${BACKEND_API_URL}/v1/hyperparams/get/`,
+        data,
+        {
+          headers: {
+            "Content-Type": "application/json",
+            token: `${token}`,
+          },
+        }
```
Comment (bamblebam) [inline FrontEndApp/v1-react/src/Components/MainApp/HomeService.js:16]: I see this code snippet is being used a lot of time in this file with just variation in the url. Can this be converted to a wrapper or something to reduce code redundancy ?

Comment (therajtiwari) [reply FrontEndApp/v1-react/src/Components/MainApp/HomeService.js:16]: > I see this code snippet is being used a lot of time in this file with just variation in the url. Can this be converted to a wrapper or something to reduce code redundancy ?

Yep we can look into this

**Pull request merged**